var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -683,16 +683,16 @@ def cmake_args(self):
                 # hence the test to see if the version starts with "flang".
                 targets.append('CppBackend')
 
-            if 'x86' in spec.architecture.target.lower():
+            if 'x86' in spec.target.family:
                 targets.append('X86')
-            elif 'arm' in spec.architecture.target.lower():
+            elif 'arm' in spec.target.family:
```

Comment:
```suggestion
            elif spec.target.family == 'arm':
```
ARM 32-bit is the only case I am unsure, as there seem to be different names for them. You can go with the change above, if anything is missing we'll fix it in another PR.